libp2p/NodeTable.cpp

```diff
@@ -220,32 +220,25 @@ vector<shared_ptr<NodeEntry>> NodeTable::nearestNodeEntries(NodeID _target)
 	static unsigned lastBin = s_bins - 1;
 	unsigned head = distance(m_node.id, _target);
 	unsigned tail = head == 0 ? lastBin : (head - 1) % s_bins;
-	
+
 	map<unsigned, list<shared_ptr<NodeEntry>>> found;
-	unsigned count = 0;
-	
+
 	// if d is 0, then we roll look forward, if last, we reverse, else, spread from d
 	if (head > 1 && tail != lastBin)
-		while (head != tail && head < s_bins && count < s_bucketSize)
+		while (head != tail && head < s_bins)
 		{
 			Guard l(x_state);
 			for (auto const& n: m_state[head].nodes)
 				if (auto p = n.lock())
 				{
-					if (count < s_bucketSize)
-						found[distance(_target, p->id)].push_back(p);
-					else
-						break;
+					found[distance(_target, p->id)].push_back(p);
 				}
-			
-			if (count < s_bucketSize && tail)
+
+			if (tail > 0)
 				for (auto const& n: m_state[tail].nodes)
 					if (auto p = n.lock())
 					{
-						if (count < s_bucketSize)
-							found[distance(_target, p->id)].push_back(p);
-						else
-							break;
+						found[distance(_target, p->id)].push_back(p);
```

Comment:
Presuming all tests pass, this should be:

    if (!!p->endpoint && p->endpoint.isAllowed() && count++ < s_bucketSize)
        found[distance(_target, p->id)].push_back(p);
    else
        break;